apps/src/lib/client/utils.rs

```diff
@@ -154,7 +121,69 @@ pub async fn join_network(
         } else {
             (PathBuf::from_str(".").unwrap(), false)
         };
-    archive.unpack(&unpack_dir).unwrap();
+
+    let chain_id = match (chain_id, release_archive) {
+        (None, None) | (Some(_), Some(_)) => {
+            eprintln!(
+                "Exactly one of --chain-id or --release-archive should have \
+                 been provided"
+            );
+            cli::safe_exit(1)
+        }
```

Comment:
Not great. It would be good if we could use an enum instead of two `Option` types, of which we know exactly one will be not `None`.

Reply:
https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.xor ?

Reply:
the `Option` types are different in this case (`Option<ChainId>` vs `Option<PathBuf>`)